### PR TITLE
test-runner image spring cleaning

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

- Removes k8s.io/test-infra dependency
- Clean unused package
- Reorganize some stuff

This keeps docker-in-docker setup in or now as 2 jobs on our configuration are using it.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by vdemeester** [comment]
:scream:

**Comment by vdemeester** [comment]
I think it's ready. The mario-test fails because the way the `e2e` script works.. but it does setup a cluster, etc..

**Comment by vdemeester** [comment]
I think we can go with this, and slowly switch some jobs to `:latest` again :angel:

File: tekton/images/test-runner/runner.sh [hunk 1]
```diff
@@ -89,7 +89,36 @@ export SOURCE_DATE_EPOCH
...
+    case $1 in
+        # FIXME(vdemeester) Remove those
+        --scenario=*) echo "scenario";;
+        --clean) echo "clean";;
+        --job=*) echo "job";;
+        --repo=*) echo "repo";;
+        --root=*) echo "root";;
+        --upload=*) echo "upload";;
+        --service-account=*)
+            echo "do something with it"
```
Comment (afrittoli) [inline tekton/images/test-runner/runner.sh:103]: NIT: I'm not sure we need this comment?

File: tekton/images/test-runner/Dockerfile [hunk 2]
```diff
@@ -215,9 +181,13 @@ RUN gem install mdl -v 0.5.0
 RUN python -m pip install yamllint
 
 # Temporarily add test-infra to the image to build custom tools
-RUN mkdir -p /go/src/github.com/knative/ && git clone https://github.com/knative/test-infra.git /go/src/github.com/knative/test-infra && \
-    make -C /go/src/github.com/knative/test-infra/tools/githubhelper && \
-    cp /go/src/github.com/knative/test-infra/tools/githubhelper/githubhelper /usr/local/bin && \
+RUN mkdir -p /${GOPATH}/src/github.com/knative/ && git clone https://github.com/knative/test-infra.git /${GOPATH}/src/github.com/knative/test-infra && \
```
Comment (afrittoli) [inline tekton/images/test-runner/Dockerfile:184]: We could build this is a separate container like we do for `kubetest` ?

File: tekton/images/test-runner/runner.sh [hunk 3]
```diff
@@ -89,7 +89,36 @@ export SOURCE_DATE_EPOCH
 
 # actually start bootstrap and the job
 set -o xtrace
+
```
Comment (afrittoli) [inline tekton/images/test-runner/runner.sh:92]: NIT: Perhaps we could have an heading echo, like "=== Running ./runner.sh backward compatibility test runner  === "

File: tekton/images/test-runner/runner.sh (hunk 1 above)
Comment (vdemeester) [reply tekton/images/test-runner/runner.sh:103]: good point, I'll need to remove that..

File: tekton/images/test-runner/Dockerfile (hunk 2 above)
Comment (vdemeester) [reply tekton/images/test-runner/Dockerfile:184]: indeed :wink:

**Pull request merged**